cpp/src/Glacier2/Glacier2Router.cpp

```diff
@@ -430,6 +430,13 @@ RouterService::initializeCommunicator(
         initData.properties->setProperty("Glacier2.Client.Connection.InactivityTimeout", "0");
     }
 
+    // Turn-off the inactivity timeout for outgoing connections unless the application sets this property.
+    // This is necessary for connections to session managers, like the session managers hosted by the IceGrid registry.
+    if (initData.properties->getProperty("Ice.Connection.Client.InactivityTimeout").empty())
```

Comment:
I think a better / more consistent approach here is to have users set this property when they select an IceGrid session manager in Glacier2 (and naturally document it).

Reply:
Fixed as suggested.

Reply:
I'm a bit concerned that they might not know about this or really understand what to do. IMO a per proxy setting would be nicer.

Reply:
I am not in favor of adding new proxy properties:
https://doc.zeroc.com/ice/3.7/property-reference/proxy-properties

especially here, it would be a proxy property that requires a connection with this special InactivityTimeout value (similar to connectionId or connection timeout in 3.7).